draft-ietf-httpbis-message-signatures.md

```diff
@@ -1008,25 +998,99 @@ HTTP/1.1 503 Service Unavailable
+Content-Type: application/json
+Content-Length: 62
+Content-Digest: sha-512=:0Y6iCBzGg5rZtoXS95Ijz03mslf6KAMCloESHObfwn\
+  HJDbkkWWQz6PhhU9kxsTbARtY2PTBOzq24uJFpHsMuAg==:
+Signature-Input: reqres=("@status" "content-digest" "content-type" \
+  "@authority";req "@method";req "@path";req "@query";req \
+  "content-digest";req "content-type";req "content-length";req)\
+  ;created=1618884479;keyid="test-key-ecc-p256"
+Signature: reqres=:zU7zd1MN56WapeNxfVNleCx5rFxBhBcZngnX4d+MurOk3tNu\
+  3rFfTFnwhglZH8qNBoygvhVMfQq9wIvLqyVNog==:
 
 {"busy": true, "message": "Your call is very important to us"}
 ~~~
 
 Note that the ECDSA algorithm in use here is non-deterministic, meaning a different signature value will be created every time the algorithm is run. The signature value provided here can be validated against the given keys, but newly-generated signature values are not expected to match the example. See {{security-nondeterministic}}.
 
-Since the signature component values from the request are not repeated in the response, the requester MUST keep the original message component values around long enough to validate the signature of the response that uses this component identifier. In most cases, this means the requester needs to keep the original request message around, since the signer could choose to include any portions of the request in its response, according to the needs of the application. Since it is possible for an intermediary to alter a request message before it is processed by the server, applications need to take care not to sign such altered values as the client would not be able to validate the resulting signature.
+Applications signing a response to a signed request SHOULD sign all of the components of the request signature value to provide sufficient coverage and protection against a class of collision attacks, as discussed in {{security-sign-signature}}. The server in this example has included all components listed in the Signature-Input of the client's signature on the request in the response signature, in addition to components of the response.
 
-Applications needing this type of binding have to sign sufficient portion the request to ensure that it is uniquely tied to the response. Signing the signature value of a signed request alone does not provide sufficient coverage in most cases, as discussed in {{security-sign-signature}}.
+While it is syntactically possible to sign another signature on a message, applications SHOULD NOT sign the Signature and Signature-Input fields of the request, as shown in this example. Doing so does not provide transitive coverage of covered components and is susceptible to attack as discussed in {{security-sign-signature}}. In some applications, signing these fields of a request can indicate to the client that the server has processed an input signature, but such applications need to carefully specify such usage in addition to ensuring sufficient coverage of the messages.
```

Comment:
```suggestion
While it is syntactically possible to include a signature component from a request as an input to a response signature, this is NOT RECOMMENDED. This is because signatures of signatures do not provide the transitive coverage of covered components as one might expect and is susceptible to various attacks as discussed in {{security-sign-signature}}.
```

Rationale: I think NOT RECOMMENDED is a bit clearer than SHOULD NOT in this case. I worry the last sentence muddies the water a little bit so I think the text is stronger without it.